src/Umbraco.Web.UI.Client/src/views/components/editor/umb-editor-content-header.html

```diff
@@ -40,18 +40,18 @@
                     </ng-form>
 
                     <a ng-if="content.variants.length > 0 && hideChangeVariant !== true" class="umb-variant-switcher__toggle" href="" ng-click="vm.dropdownOpen = !vm.dropdownOpen" ng-class="{'--error': vm.errorsOnOtherVariants}">
-                        <span>{{vm.currentVariant.language.name}}</span>
+                        <span ng-bind="getVariantDisplayName(vm.currentVariant)"></span>
                         <ins class="umb-variant-switcher__expand" ng-class="{'icon-navigation-down': !vm.dropdownOpen, 'icon-navigation-up': vm.dropdownOpen}">&nbsp;</ins>
                     </a>
-
+                    
                     <span ng-if="hideChangeVariant" class="umb-variant-switcher__toggle">
-                        <span>{{vm.currentVariant.language.name}}</span>
+                        <span ng-bind="getVariantDisplayName(variant)"></span>
                     </span>
 
                     <umb-dropdown ng-if="vm.dropdownOpen" style="min-width: 100%; max-height: 250px; overflow-y: auto; margin-top: 5px;" on-close="vm.dropdownOpen = false" umb-keyboard-list>
-                        <umb-dropdown-item class="umb-variant-switcher__item" ng-class="{'--current': variant.active, '--not-allowed': variantIsOpen(variant.language.culture), '--error': variantHasError(variant.language.culture)}" ng-repeat="variant in content.variants">
+                        <umb-dropdown-item class="umb-variant-switcher__item" ng-class="{'--current': variant.active, '--not-allowed': variantIsOpen(variant), '--error': variantHasError(variant.language.culture)}" ng-repeat="variant in content.variants">
```

Comment:
variantHasError is still only checking culture, have you dealt with validation for segments at this stage?